packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -664,7 +664,7 @@ function validateClassAttr(
 
     const elmClassName = getAttribute(elm, 'class');
 
-    if (!isUndefined(className) && String(className) !== elmClassName) {
+    if (!isUndefined(className) && String(className) !== elmClassName && className !== '') {
```

Comment:
This isn't totally right. If the client class name is `''` but the server class name is `'foo'` then there will be no mismatch when there should be.

Reply:
See: https://github.com/salesforce/lwc/pull/4698

Reply:
Thanks! I figured some case wasn't covered here.